### PR TITLE
retrieve only template id

### PR DESCRIPTION
**Pull request by dmitrytulba**
# Changes

> Describe your changes in a short and concise list.

# Related issue

> Reference related issues, and use one of the [closing keywords, e.g. closes or fixes](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) to link the corresponding issue, if any.

# Additional information 

> Provide additional information for this change, if needed.

# Pre-Merge checklist

- [ ] New code is tested
- [ ] Present and new tests pass
- [ ] Documentation is updated
- [ ] The build is working without errors
- [ ] No new Sonar issues introduced
- [ ] Changelog is updated
- [ ] Code has been reviewed

### CODE REVIEW

File: service/src/main/java/org/ehrbase/dao/access/jooq/EntryAccess.java [hunk 1]
```diff
@@ -119,14 +120,28 @@ private EntryAccess(I_DomainAccess domainAccess) {
...
+    public static String getTemplateIdFromEntry(I_DomainAccess domainAccess, UUID compositionId) {
+        return fetchEntryByCompositionId(domainAccess, compositionId)
+                .map(EntryRecord::getTemplateId)
+                .filter(StringUtils::isNotBlank)
+                .orElse(null);
+    }
+
+    private static Optional<EntryRecord> fetchEntryByCompositionId(I_DomainAccess domainAccess, UUID compositionId) {
+        try {
+            return domainAccess.getContext().fetchOptional(ENTRY, ENTRY.COMPOSITION_ID.eq(compositionId));
```
Comment (dmitrytulba) [inline service/src/main/java/org/ehrbase/dao/access/jooq/EntryAccess.java:132]: get only template id instead  of whole entry

Comment (dmitrytulba) [reply service/src/main/java/org/ehrbase/dao/access/jooq/EntryAccess.java]: done

File: service/src/main/java/org/ehrbase/dao/access/jooq/EntryAccess.java [hunk 2]
```diff
@@ -119,6 +115,17 @@ private EntryAccess(I_DomainAccess domainAccess) {
         super(domainAccess);
     }
 
+    public static String fetchTemplateIdByCompositionId(I_DomainAccess domainAccess, UUID compositionId) {
```
Comment (HolgerReiseVSys) [inline service/src/main/java/org/ehrbase/dao/access/jooq/EntryAccess.java:118]: `fetchOptional()` could be used, or in this case also `fetchOne(int fieldIndex)`

File: CHANGELOG.md [hunk 3]
```diff
@@ -15,6 +15,7 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
  ### Changed
 - use caffeine cache instead of ehcache as ehcache has unnecessary blocking([#1007](https://github.com/ehrbase/ehrbase/pull/1007))
 - remove unnecessary DB queries([#1007](https://github.com/ehrbase/ehrbase/pull/1007))
+- Acquire the template identification only instead of composition.([#1042](https://github.com/ehrbase/ehrbase/pull/1042))
```
Comment (HolgerReiseVSys) [inline CHANGELOG.md:18]: Seems overly specific , maybe just mention performance improvements of the audit feature?

**Review by HolgerReiseVSys — CHANGES_REQUESTED** [review]

File: service/src/main/java/org/ehrbase/dao/access/jooq/EntryAccess.java (hunk 2 above)
Comment (dmitrytulba) [reply service/src/main/java/org/ehrbase/dao/access/jooq/EntryAccess.java:118]: done

File: CHANGELOG.md (hunk 3 above)
Comment (dmitrytulba) [reply CHANGELOG.md:18]: done

**Review by HolgerReiseVSys — APPROVED** [review]

**Pull request merged**